### DISCUSSION TITLE
在apisix集群中如何维护和使用服务发现？

### DISCUSSION THREAD

**Discussion by FinerKeysen** [apache/apisix#8397] [Q&A] [answered]
apisix version: 2.13.3

如果我搭建了一个三节点的apisix集群，我想在这个集群中使用nacos服务发现能力，那么我是否需要在每个apisix节点上都写入discovery.nacos的配置？

如果只在其中一个节点中配置服务发现，那么在其他节点能生效吗？

**Comment by tzssangglass** [comment, marked as answer]
> 那么我是否需要在每个apisix节点上都写入discovery.nacos的配置？

yes

> 如果只在其中一个节点中配置服务发现，那么在其他节点能生效吗？

no

**Comment by showcup** [comment]
apisix节点之间是无状态的，配置是无法自动共享的，需要每个apisix节点做相同的配置
